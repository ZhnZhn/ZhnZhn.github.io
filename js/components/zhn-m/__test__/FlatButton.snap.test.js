/**
 * @jest-environment jsdom
 */
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));
var _FlatButton = _interopRequireDefault(require("../FlatButton"));
var _jsxRuntime = require("react/jsx-runtime");
const _crProps = option => ({
  caption: "Load",
  title: "Click to load",
  hotKey: "l",
  onClick: jest.fn(),
  ...option
});
describe('FlatButton Snap', () => {
  test('should renders correctly', () => {
    const tree = _reactTestRenderer.default.create(/*#__PURE__*/(0, _jsxRuntime.jsx)(_FlatButton.default, {
      ..._crProps()
    })).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should renders correctly without title', () => {
    const tree = _reactTestRenderer.default.create(/*#__PURE__*/(0, _jsxRuntime.jsx)(_FlatButton.default, {
      ..._crProps({
        title: void 0
      })
    })).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=FlatButton.snap.test.js.map