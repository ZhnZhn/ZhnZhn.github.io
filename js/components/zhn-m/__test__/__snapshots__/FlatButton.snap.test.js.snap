// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FlatButton Snap should renders correctly 1`] = `
<button
  className="bt-flat bold-16"
  onClick={[Function]}
  title="Click to load"
  type="button"
>
  <span
    className="bt-flat__caption"
  >
    Load
  </span>
</button>
`;

exports[`FlatButton Snap should renders correctly without title 1`] = `
<button
  className="bt-flat bold-16"
  onClick={[Function]}
  type="button"
>
  <span
    className="bt-flat__caption"
  >
    Load
  </span>
</button>
`;
